### PR TITLE
code change: Improve code around flavors

### PR DESCRIPTION
**Pull request by akrejcir**
**What this PR does / why we need it**:
Extract adding flavor name annotations into a separate function.

**Release note**:
```release-note
None
```

### CODE REVIEW

File: pkg/flavor/flavor.go [hunk 1]
```diff
@@ -36,6 +34,8 @@ type methods struct {
 	clusterFlavorStore cache.Store
 }
 
+var _ Methods = &methods{}
```
Comment (xpivarc) [inline pkg/flavor/flavor.go:37]: I think this deserves an explanation/note in the commit message.

File: pkg/virt-controller/watch/vm.go [hunk 2]
```diff
@@ -1134,7 +1134,9 @@ func (c *VMController) applyFlavorToVmi(vm *virtv1.VirtualMachine, vmi *virtv1.V
 		return nil
 	}
 
-	conflicts := c.flavorMethods.ApplyToVmi(k8sfield.NewPath("spec"), flavorProfile, vm, vmi)
+	flavor.AddFlavorNameAnnotations(vm, vmi)
```
Comment (xpivarc) [inline pkg/virt-controller/watch/vm.go:1137]: What's the reasoning for moving this call out of `ApplyToVmi`?

File: pkg/flavor/flavor.go (hunk 1 above)
Comment (akrejcir) [reply pkg/flavor/flavor.go:37]: This is a golang idiom. It is a dummy variable, so the compiler makes sure that `*methods` struct implements `Methods` interface. It is helpful when refactoring interfaces, so we don't forget to update the structs that implement them.

**Comment by akrejcir** [comment]
@xpivarc do you know who could review and approve this?

**Comment by akrejcir** [comment]
/cc @jean-edouard @vladikr 
Hi, could you please review this PR, or mention someone who can?

**Review by jean-edouard — COMMENTED** [review]
/approve

**Comment by lyarwood** [comment]
@vladikr @davidvossel apologies for the direct ping but could one of you lgtm this please so we can start posting some additional early parts of the flavor series.

**Comment by davidvossel** [comment]
> @vladikr @davidvossel apologies for the direct ping but could one of you lgtm this please so we can start posting some additional early parts of the flavor series.

@lyarwood i think you can add lgtm, you would just need us for an approve.

**Comment by lyarwood** [comment]
> > @vladikr @davidvossel apologies for the direct ping but could one of you lgtm this please so we can start posting some additional early parts of the flavor series.
> 
> @lyarwood i think you can add lgtm, you would just need us for an approve.

Apologies I thought I had to be listed in the OWNERS file as a code-reviewer for that to work and not just a member of the org.

/lgtm

**Comment by lyarwood** [comment]
`Error in download: java.io.IOException: Error downloading [http://mirror.centos.org/centos/8-stream/BaseOS/x86_64/os/Packages/selinux-policy-targeted-3.14.3-95.el8.noarch.rpm]`  unrelated failure.

/retest

**Pull request merged**